File: client/src/types/commitment.ts
export interface CommitmentDetails {
  name: string;
  commitments: Commitment[];
  assetClasses: AssetClass[];
}

export interface Commitment {
  id: number;
  assetClass: string;
  amount: string;
  currency: string;
}

export interface AssetClass {
  name: string;
  totalAmount: string;
}

File: Domain/Commitment.cs
namespace Domain
{
    public class Commitment
    {
        public int Id { get; set; }

        public int InvestorId { get; set; }

        public string AssetClass { get; set; } = string.Empty;

        public decimal Amount { get; set; }

        public string Currency { get; set; } = "GBP";
        
        public Investor Investor { get; set; }
    }
}